src/eko/msbar_masses.py

```diff
@@ -328,52 +334,71 @@ def evolve(
             )
             matching = 1.0
             for pto in range(1, strong_coupling.order[0]):
-                for l in range(pto + 1):
+                for logpow in range(pto + 1):
                     as_thr = strong_coupling.a(
-                        seg.q2_to / fact_to_ren, seg.q2_to, seg.nf - shift + 4
+                        seg.q2_to / xif2, seg.q2_to, seg.nf - shift + 4
                     )[0]
-                    matching += as_thr**pto * L**l * m_coeffs[pto, l]
+                    matching += as_thr**pto * L**logpow * m_coeffs[pto, logpow]
             ker_evol *= matching
         ev_mass *= ker_evol
     return m2_ref * ev_mass
 
 
-def compute(theory_card):
-    r"""
-    Compute the |MSbar| masses solving the equation :math:`m_{\bar{MS}}(\mu) = \mu`
-    for each heavy quark and consistent boundary contitions.
+def compute(
+    masses_ref: HeavyQuarkMasses,
+    couplings: CouplingsRef,
+    order: Order,
+    evmeth: CouplingEvolutionMethod,
+    matching: MatchingScales,
+    xif2: float = 1.0,
+):
+    r"""Compute the |MSbar| masses.
+
+    Computation is performed solving the equation :math:`m_{\bar{MS}}(\mu) =
+    \mu` for each heavy quark and consistent boundary contitions.
 
     Parameters
     ----------
-        theory_card: dict
-            theory run card
+    nf_ref :
+        number of active flavor at reference scale
+    mu2_ref :
+        reference scale squared (a.k.a. :math:`Q^2_{ref}`)
+    xif2 :
+        squared ratio of factorization to central scale
 
     Returns
     -------
-        masses: list
-            list of |MSbar| masses squared
+    masses : list
+        list of |MSbar| masses squared
+
     """
     # TODO: sketch in the docs how the MSbar computation works with a figure.
-    nfa_ref = theory_card["nfref"]
-
-    q2_ref = np.power(theory_card["Qref"], 2)
-    masses = np.concatenate((np.zeros(nfa_ref - 3), np.full(6 - nfa_ref, np.inf)))
-    fact_to_ren = theory_card["fact_to_ren_scale_ratio"] ** 2
+    mu2_ref = couplings.alphas.scale**2
```

Comment:
oookay, so the statement "inside eko all scales are squared" is apparently slightly more convoluted: "inside eko" really means "inside the respective function bodies and classes", but not inside the runcard objects (such as `couplings.alphas.scale`) which are now passed around. To be explicit: "inside eko" does __not__ mean "the arguments of functions" - do you confirm @AleCandido ?

Reply:
The difference from inside and outside is really IO: if it is user input is outside, and runcards represent the user input (since their content comes directly from deserialization).
However, we can make properties for squared scales :)